diff --git include/vigra/separableconvolution.hxx include/vigra/separableconvolution.hxx
index 7ee33c09d..8fd120602 100644
--- include/vigra/separableconvolution.hxx
+++ include/vigra/separableconvolution.hxx
@@ -1409,10 +1409,10 @@ class Kernel1D
         {}
 
         ~InitProxy()
-#ifndef _MSC_VER
-            throw(PreconditionViolation)
-#elif _MSC_VER >= 1900
+#if _MSC_VER >= 1900 || __cplusplus >= 201103L
             noexcept(false)
+#else
+            throw(PreconditionViolation)
 #endif
         {
             vigra_precondition(count_ == 1 || count_ == sum_,
diff --git include/vigra/stdconvolution.hxx include/vigra/stdconvolution.hxx
index 76d61a53f..657247259 100644
--- include/vigra/stdconvolution.hxx
+++ include/vigra/stdconvolution.hxx
@@ -792,10 +792,10 @@ public:
         {}
 
         ~InitProxy()
-#ifndef _MSC_VER
-            throw(PreconditionViolation)
-#elif _MSC_VER >= 1900
+#if _MSC_VER >= 1900 || __cplusplus >= 201103L
             noexcept(false)
+#else
+            throw(PreconditionViolation)
 #endif
         {
             vigra_precondition(count_ == 1 || count_ == sum_,
