diff --git config/run_test.bat.in config/run_test.bat.in
index f0a60377c..fc66200a4 100644
--- config/run_test.bat.in
+++ config/run_test.bat.in
@@ -5,7 +5,11 @@ if [%1] NEQ [] (
     set CONFIGURATION=Release
 )
 
-echo Testing configuration '%CONFIGURATION%'
+if "%CONFIGURATION%" EQU "." (
+    echo Testing configuration '@CMAKE_BUILD_TYPE@'
+) else (
+    echo Testing configuration '%CONFIGURATION%'
+)
 
 set PATH=@EXTRA_PATH@%PATH%
 
diff --git vigranumpy/test/CMakeLists.txt vigranumpy/test/CMakeLists.txt
index fd8e361e4..d72751068 100644
--- vigranumpy/test/CMakeLists.txt
+++ vigranumpy/test/CMakeLists.txt
@@ -153,11 +153,7 @@ IF(NOT PYTHON_NOSETESTS_NOT_FOUND)
     # variable CMAKE_CFG_INTDIR contains a dot '.' for a Windows nmake build, or
     #                           '$(OutDir)' for a VisualStudio build (OutDir will be
     #                           set by VS at build time)
-    IF(NOT CMAKE_CFG_INTDIR STREQUAL ".")
-        SET(VIGRA_CONFIGURATION ${CMAKE_CFG_INTDIR})
-    ELSE()
-        SET(VIGRA_CONFIGURATION)
-    ENDIF()
+    SET(VIGRA_CONFIGURATION ${CMAKE_CFG_INTDIR})
 
     IF(AUTOEXEC_TESTS)
         add_custom_command(
